presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -226,4 +227,155 @@ private void checkEncodeInputValidity(List<Integer> input, int maximumBits)
 
         checkArgument((totalBitLength <= maximumBits), format("The z-address type specified is not large enough to hold %d bits.", totalBitLength));
     }
+
+    /**
+     * Searches for and outputs the addresses within certain ranges in each dimension.
+     *
+     * @param minimumValues the list of minimum values of each dimension
+     * @param maximumValues the list of maximum values of each dimension
+     * @return the list of addresses within the given value ranges
+     */
+    public List<Long> zOrderSearchCurve(List<Integer> minimumValues, List<Integer> maximumValues)
```

Comment:
For the input, is it possible to have a wrapper class (say, `Range`)? This way we don't need the implicit assumption that these two lists has a index-based mapping.